dy_lesson_04/lesson4.ipynb

```diff
@@ -0,0 +1,994 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Dictionaries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_dict1 = dict()\n",
+    "my_dict2 = {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "dict"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "type(my_dict1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "dict"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "type(my_dict2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'Bob': 10, 'Alice': 15, 'DY': 7}\n",
+      "{'Bob': [10, 12], 'Alice': {'Hey': [15, 7]}, 'DY': 7}\n"
+     ]
+    }
+   ],
+   "source": [
+    "example1 = {\"Bob\" : 10, \"Alice\" : 15, \"DY\" : 7}\n",
+    "example2 = {}\n",
+    "example2[\"Bob\"] = [10, 12]\n",
+    "example2[\"Alice\"] = {\"Hey\" : [15, 7]}\n",
+    "example2[\"DY\"] = 7\n",
+    "print(example1)\n",
+    "print(example2)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dict_keys(['Bob', 'Alice', 'DY'])\n",
+      "dict_values([10, 15, 7])\n",
+      "Keys\n",
+      "Bob\n",
+      "Alice\n",
+      "DY\n",
+      "\n",
+      "Values\n",
+      "10\n",
+      "15\n",
+      "7\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(example1.keys())\n",
+    "print(example1.values())\n",
+    "\n",
+    "print(\"Keys\")\n",
+    "for key in example1.keys():\n",
+    "    print(key)\n",
+    "\n",
+    "print()\n",
+    "\n",
+    "print(\"Values\")\n",
+    "for value in example1.values():\n",
+    "    print(value)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dict_items([('Bob', 10), ('Alice', 15), ('DY', 7)])\n",
```

Comment:
This is an example of a comment in a pull request for a review

Reply:
What should be fixed? Please elaborate more

Reply:
this looks good